internal/internal_activity.go

```diff
@@ -302,98 +302,131 @@ func isActivityContext(inType reflect.Type) bool {
 	return inType != nil && inType.Implements(contextElem)
 }
 
-func validateFunctionAndGetResults(f interface{}, values []reflect.Value, dataConverter converter.DataConverter) (*commonpb.Payloads, error) {
-	resultSize := len(values)
-
-	if resultSize < 1 || resultSize > 2 {
-		fnName, _ := getFunctionName(f)
-		return nil, fmt.Errorf(
-			"the function: %v signature returns %d results, it is expecting to return either error or (result, error)",
-			fnName, resultSize)
+func setActivityParametersIfNotExist(ctx Context) Context {
```

Comment:
nit: s/Parameters/Options for consistency?
Same for the local var names

Reply:
This is not my code, it remains unchanged. Git diff is struggling with this file.